### PR TITLE
Make window object the default parameter instead of forcing it

### PR DESCRIPTION
**Pull request by louisgv**
Related issue: #2 
Refactor window into the argument. This allows consumer of this module to use their own websocket client implementation.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ const events = require('events')
 const through2 = require('through2')
 const inherits = require('inherits')
 
-function SignalhubWs (app, urls) {
+function SignalhubWs (app, urls, wsClass = window.WebSocket) {
```
Comment (soyuka) [inline index.js]: May we declare `const WebSocket = window && window.WebSocket ? window.WebSocket : null`. This way we avoid errors when window is undefined.

Then `(app, urls, ws = WebSocket)` 

And if no implementation is given (ie `ws === null`, let's do `throw TypeError('No WebSocket class given.')`?

**Comment by ColdSauce** [comment]
@soyuka we've made the modifications you were suggesting and also added browser and server tests. Please take a look when you can!

**Comment by soyuka** [comment]
thanks guys!

**Comment by soyuka** [comment]
landed in 1.0.4

**Pull request merged**